client_auto.go

```diff
@@ -0,0 +1,131 @@
+package hdfs
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/colinmarc/hdfs/rpc"
+)
+
+type Property struct {
+	Name  string `xml:"name"`
+	Value string `xml:"value"`
+}
+
+type Result struct {
+	Property []Property `xml:"property"`
+}
+
+type NameNode struct {
+	Host string
+	Port int
+}
+
+type nameNodeError struct {
```

Comment:
There's no need for this to be a struct. it can just be `var ErrUnresolvedNamenode = errors.New("Couldn't find a namenode address in any config.")`